x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/action/EsqlCapabilities.java

```diff
@@ -549,6 +549,11 @@ public enum Cap {
          */
         JOIN_LOOKUP_V7(Build.current().isSnapshot()),
 
+        /**
+         * LOOKUP JOIN with index mode verification
+         */
+        JOIN_LOOKUP_VERIFY_MODE(JOIN_LOOKUP_V7.isEnabled()),
+
```

Comment:
Nit: this work is part of Lookup so I would reuse the existing capability (since it's compatible) or bump JOIN_LOOKUP to V8.

Reply:
When executed in BWC, the YAML test may fail as they don't have this verification (?).
The new cap, was to avoid conflicts, but as it's a minor thing, I increased the version instead now